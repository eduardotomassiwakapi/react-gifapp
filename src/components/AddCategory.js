import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories, /*limit, setLimit*/ }) => {

    const [inputCategoryValue, setInputCategoryValue] = useState('');
    //const [inputLimitValue, setInputLimitValue] = useState(limit)
    
    const handleCategoryInputChange = (e) => {
        //console.log("handleCategoryInputChange="+e.target.value);
        setInputCategoryValue(e.target.value);
    }
    // const handleLimitInputChange = (e) => {
    //     setInputLimitValue(e.target.value);
    // }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputCategoryValue.trim().length > 2) {
            //console.log("handleSubmit="+inputCategoryValue);
            setCategories(oldCategories => [inputCategoryValue, ...oldCategories,]);
            //setLimit(inputLimitValue);
            setInputCategoryValue('');
        }
    }


    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text" 
                value={inputCategoryValue}
                onChange={ (e) => handleCategoryInputChange(e)} />
            {/* <input type="number" 
            value={limit} onChange={ (e) => handleLimitInputChange(e)} /> */}
        </form>
    )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
   //setLimit: PropTypes.func.isRequired
};
