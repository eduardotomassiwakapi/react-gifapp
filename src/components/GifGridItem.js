import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({title,url}) => {
    //console.log({id,title,url});
    return (
        <div className='card animate__animated animate__fadeIn'>
            <h4>{title}</h4>
            <img src={url} alt={title} />
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};