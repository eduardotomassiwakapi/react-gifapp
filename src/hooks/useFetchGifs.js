import { useEffect, useState } from 'react'
import { fetchGifs } from '../helpers/fetchGifs';

export const useFetchGifs = (apiUrl) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        fetchGifs(apiUrl)
            .then( images => {
                // setTimeout( () => {
                    //console.log(images);
                    setState({
                        data: images,
                        loading: false
                    })
                // }, 1000);
            })
    }, [apiUrl] );

    
    return state;
}
 