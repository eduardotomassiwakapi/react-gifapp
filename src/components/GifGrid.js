// import React, { useState, useEffect } from 'react'
// import { fetchGifs } from '../helpers/fetchGifs';
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifGrid = ({ category, limit }) => {

    const apiKey = "1H3c5Ybz7bYQgUPoGaTiMK228pvRc0nK";
    const apiUrl = `https://api.giphy.com/v1/gifs/search?rating=r&q=${encodeURI(category)}&limit=${limit}&api_key=${apiKey}`;
    const {data: images, loading} = useFetchGifs(apiUrl);

    return ( 
        <>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid">
                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
