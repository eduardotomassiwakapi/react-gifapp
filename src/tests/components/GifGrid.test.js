import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")

describe('<GifGrid />', () => {

    // MOCKS
    // useFetchGifs = jest.fn().mockResolvedValue({
    //     data: [
    //       {
    //         id: "Id",
    //         title: 'Title',
    //         url: 'URL'
    //       }
    //     ]
    //   });
    // useFetchGifs.mockReturnValue({
    //     data: [
    //         {
    //             id: "item.id",
    //             title: "item.title",
    //             url: "item.images?.fixed_height.url"
    //         }
    //     ],
    //     loading: true
    // });

    //BEFORE TESTS
    const category = 'Test'
    const limit = 20;
    // let wrapper = shallow(<GifGrid category={category} limit={limit} />);
    // beforeEach(() => {
    //     jest.clearAllMocks();
    //     wrapper = shallow(<GifGrid category={category} limit={limit} />);
    // })


    test('Should render properly with Loading...', () => {

        useFetchGifs.mockReturnValue({
            data: [ ],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} limit={limit} />);
        
        expect(wrapper.find('h3').text()).toBe(category);
        expect(wrapper).toMatchSnapshot();
    });

    test('Should show items when images get loaded from useFetchGifs', () => {
        const gifs = [{
            id: "item.id",
            title: "item.title",
            url: "item.images?.fixed_height.url"
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false //not loading any more
        });
        const wrapper = shallow(<GifGrid category={category} limit={limit} />);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

        //expect(wrapper).toMatchSnapshot();
    });
    
})
