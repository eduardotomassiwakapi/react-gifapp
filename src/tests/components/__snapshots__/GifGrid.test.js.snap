// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> Should render properly with Loading... 1`] = `
<Fragment>
  <h3>
    Test
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
